node/src/database/db_migrations.rs

```diff
@@ -0,0 +1,502 @@
+// Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::database::connection_wrapper::ConnectionWrapper;
+use crate::database::db_initializer::CURRENT_SCHEMA_VERSION;
+use crate::sub_lib::logger::Logger;
+use masq_lib::utils::ExpectDecent;
+use regex::Regex;
+use std::fmt::Debug;
+use std::ops::{Deref, Not};
+use std::slice::Iter;
+
+const THE_EARLIEST_ENTRY_IN_THE_LIST_OF_DB_MIGRATIONS: &str = "0.0.10";
+
+pub trait DbMigrator {
+    fn migrate_database(
+        &self,
+        mismatched_schema: &str,
+        conn: Box<dyn ConnectionWrapper>,
+    ) -> Result<(), String>;
+}
+
+pub struct DbMigratorReal {
+    logger: Logger,
+}
+
+impl DbMigrator for DbMigratorReal {
+    fn migrate_database(
+        &self,
+        mismatched_schema: &str,
+        conn: Box<dyn ConnectionWrapper>,
+    ) -> Result<(), String> {
+        self.make_updates(mismatched_schema, Self::list_of_existing_updates(), conn)
+    }
+}
+
+impl Default for DbMigratorReal {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+trait MigrateDatabase: Debug {
+    fn migrate<'a>(&self, conn: &mut Box<dyn ConnectionWrapper + 'a>) -> rusqlite::Result<()>;
+    fn version_compatibility(&self) -> &str;
+}
+
+//define your update here and add it to this list: 'list_of_existing_updates()'
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+#[allow(non_camel_case_types)]
+struct Migrate_0_0_10_to_0_0_11;
+
+impl MigrateDatabase for Migrate_0_0_10_to_0_0_11 {
```

Comment:
I think I'd prefer a noun for the name of this trait, and verb for the name of its method. I like "migrate" for the method, but how about "DatabaseMigration" for the trait?